### PR TITLE
Feature/issue 742 exclude our devices from analytics

### PR DESCRIPTION
**Pull request by rasmus-storjohann-PG**

### CODE REVIEW

**Comment by rasmus-storjohann-PG** [comment]
@ian-lin-PG @tomy-pg @bensinjin PR is up for you to look at. Quite a bit of renames, and some Redux changes.

File: src/components/search/api/use_fetch_lat_long_from_location.ts [hunk 1]
```diff
@@ -63,3 +68,16 @@ const getTextIfValidOrThrow = (response: Response): Promise<string> => {
 const handleError = R.curry((setLatLong: (latLong: LatLong) => void, _: string): void => {
     setLatLong(undefined);
 });
+
+export const useDisableAnalyticsOnEasterEgg = (location: string, disableAnalytics: (disable: boolean) => DisableAnalyticsAction): void => {
```
Comment (bensinjin) [inline src/components/search/api/use_fetch_lat_long_from_location.ts]: This probably isn't a hook you'll use more than once so probably can just put it in the file it's used in.

Comment (bensinjin) [inline src/components/search/api/use_fetch_lat_long_from_location.ts]: Dunno if this is really reusable so maybe should just live in the file it's used in.

Comment (rasmus-storjohann-PG) [reply src/components/search/api/use_fetch_lat_long_from_location.ts]: Fixed

**Pull request merged**